File: src/templates_py/templates_py/receiver.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class Receiver(Node):

    def __init__(self, node_name):
        super().__init__(node_name)
        self.sub = self.create_subscription(
            msg_type=String,
            topic='/example_topic',
            callback=self.callback,
            qos_profile=10
        )

    def callback(self, msg):
        self.get_logger().info(message=f'Receiving: {msg.data}')

def main(args=None):
    rclpy.init(args=args)
    receiver = Receiver(node_name='receiver')
    rclpy.spin(receiver)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/templates_cpp/src/receiver.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

class Receiver : public rclcpp::Node
{
    public:

    Receiver(std::string node_name) : Node(node_name)
    {
        sub = this->create_subscription<std_msgs::msg::String>
        (
            "/example_topic",
            10,
            std::bind(&Receiver::callback, this, std::placeholders::_1)
        );
    }

    private:

    void callback(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Receiving: %s", msg->data.c_str());
    }

    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto receiver = std::make_shared<Receiver>("receiver");
    rclcpp::spin(receiver);
    rclcpp::shutdown();

    return 0;
}